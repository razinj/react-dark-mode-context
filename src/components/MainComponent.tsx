import { FC } from 'react'
import logo from '../logo.svg'
import ThemeSwitch from './ThemeSwitch'

const MainComponent: FC = () => {
  return (
    <div className='font-sans flex flex-col justify-center items-center h-screen dark:bg-zinc-700 dark:text-white'>
      <img src={logo} width={200} alt='React Logo' />
      <h1 className='text-2xl'>Hello World!</h1>
      <h2>React + TailwindCSS Dark Mode App</h2>
      <div className='mt-2'>
        <ThemeSwitch />
      </div>
    </div>
  )
}

export default MainComponent
