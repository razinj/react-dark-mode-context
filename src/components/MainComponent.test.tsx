import { render } from '@testing-library/react'
import MainComponent from './MainComponent'

describe('MainComponent', () => {
  it('should match snapshot', () => {
    const { container } = render(<MainComponent />)

    expect(container).toMatchSnapshot()
  })

  it('should render component correctly', () => {
    const { getByText, getByAltText } = render(<MainComponent />)

    expect(getByAltText(/React Logo/i)).toBeInTheDocument()
    expect(getByText(/Hello World/i)).toBeInTheDocument()
  })
})
