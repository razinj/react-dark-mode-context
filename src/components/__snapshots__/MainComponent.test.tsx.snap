// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent should match snapshot 1`] = `
<div>
  <div
    class="font-sans flex flex-col justify-center items-center h-screen dark:bg-zinc-700 dark:text-white"
  >
    <img
      alt="React Logo"
      src="logo.svg"
      width="200"
    />
    <h1
      class="text-2xl"
    >
      Hello World!
    </h1>
    <h2>
      React + TailwindCSS Dark Mode App
    </h2>
    <div
      class="mt-2"
    >
      <button
        style="padding: 5px; border-radius: 5px; color: white; background: black;"
      >
        Go 
        DARK MODE
      </button>
    </div>
  </div>
</div>
`;
